### PR TITLE
Rank Rework (+Field Officer)

### PR DESCRIPTION
**Pull request by ActualRory**
:cl:
add: Field Officer Role.
add: New fluff ranks.
tweak: Existing ranks redone. 
tweak: Many demotions, some promotions.
tweak: Some roles renamed.
balance: Captain and XO have completed fencing lessons.
/:cl:

This also acts to set up ourselves for a scaling rank system. Hence several demotions and the occasional `//scaling` to indicate the current approximate rank range. Subject to change.

### CODE REVIEW

File: code/game/jobs/job/shipside.dm [hunk 1]
```diff
@@ -49,7 +50,7 @@ Godspeed, commander! And remember, you are not above the law."}
 /datum/job/command/commander/announce_entry_message(mob/living/carbon/human/H)
 	. = ..()
 	sleep(15)
-	if(H?.loc && flags_startup_parameters & ROLE_ADD_TO_MODE) 
+	if(H?.loc && flags_startup_parameters & ROLE_ADD_TO_MODE)
 		captain_announcement.Announce("All hands, Commander [H.real_name] on deck!")
```
Comment (Ghommie) [inline code/game/jobs/job/shipside.dm]:
```suggestion
		captain_announcement.Announce("All hands, Captain [H.real_name] on deck!")
```

File: code/game/jobs/job/shipside.dm [hunk 2]
```diff
@@ -91,13 +92,42 @@ Come hell or high water, you are going to be there for them."}
...
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/marine(H), WEAR_FEET)
+	H.equip_to_slot_or_del(new /obj/item/storage/backpack/marine/satchel(H), WEAR_BACK)
+	H.equip_to_slot_or_del(new /obj/item/storage/belt/gun/m4a3/vp70(H), WEAR_WAIST)
+	H.equip_to_slot_or_del(new /obj/item/storage/pouch/general/large(H), WEAR_R_STORE)
+	H.equip_to_slot_or_del(new /obj/item/storage/pouch/general/large(H), WEAR_L_STORE)
+	H.equip_to_slot_or_del(new /obj/item/device/binoculars(H), WEAR_L_HAND)
+	H.equip_to_slot_or_del(new /obj/item/clothing/suit/storage/marine/MP/RO(H), WEAR_JACKET)
+
+/datum/job/command/executive/generate_entry_message(mob/living/carbon/human/H)
+		. = {"You are the commander's 'man on the ground,' and often act on his behalf while planetside.
```
Comment (Ghommie) [inline code/game/jobs/job/shipside.dm]:
```suggestion
		. = {"You are the captain's 'man on the ground,' and often act on his behalf while planetside.
```

File: code/modules/clothing/under/marine_uniform.dm [hunk 3]
```diff
@@ -132,6 +132,15 @@
...
+	desc = "A uniform typically worn by TGMC Field Officers. It has shards of light Kevlar to help protect against stabbing weapons, bullets, and shrapnel from explosions, a small EMF distributor to help null energy-based weapons, and a hazmat chemical filter weave to ward off biological and radiation hazards."
+	icon_state = "BO_jumpsuit" //TBD: Unique Field Officer Jumpsuit
+
+/obj/item/clothing/under/marine/officer/exec/New()
+	select_gamemode_skin(/obj/item/clothing/under/marine/officer/field)
+	..()
+
 /obj/item/clothing/under/marine/officer/command
 	name = "commander uniform"
 	desc = "The well-ironed uniform of a TGMC commander. Even looking at it the wrong way could result in being court-marshalled. It has shards of light Kevlar to help protect against stabbing weapons, bullets, and shrapnel from explosions, a small EMF distributor to help null energy-based weapons, and a hazmat chemical filter weave to ward off biological and radiation hazards."
```
Comment (Ghommie) [inline code/modules/clothing/under/marine_uniform.dm]: There are quite a few istances of commander left in the code. Where appropriated, replace them with captain.

Comment (ActualRory) [reply code/modules/clothing/under/marine_uniform.dm]: Duly noted. Good spotting.

**Comment by ActualRory** [comment]
Conflicts with LaKiller's stuff should be resolved.
When I get a moment I'll give it a proper test and fix those usages of 'commander.'

File: code/game/jobs/access.dm [hunk 4]
```diff
@@ -263,29 +264,36 @@ proc/get_all_job_icons()
...
-		if("O2") . = size ? "LT " : "Lieutenant"
-		if("O3") . = size ? "LCDR " : "Lieutenant Commander"
-		if("O4") . = size ? "CDR " : "Commander"
-		if("O5") . = size ? "CPT " : "Captain"
-		if("O6") . = size ? "RADM " : "Rear Admiral"
-		if("O7") . = size ? "ADM " : "Admiral"
-		if("O8") . = size ? "FADM " : "Fleet Admiral"
-		if("O9") . = size ? "SMR " : "Sky Marshal"
-		if("WO") . = size ? "WO " : "Warrant Officer"
+		if("WY1") . = size ? (gender == "female" ? "Ms. " : "Mr. ") : "Junior Executive"
```
Comment (Rohesie) [inline code/game/jobs/access.dm]:
```suggestion
		if("NT") . = size ? (gender == "female" ? "Ms. " : "Mr. ") : "Junior Executive"
```

File: code/game/jobs/job/shipside.dm [hunk 5]
```diff
@@ -12,9 +12,10 @@ var/datum/announcement/minor/captain_announcement = new(do_newscast = 1)
 
 //Commander
 /datum/job/command/commander
-	title = "Commander"
+	title = "Captain"
+	disp_title = "Captain"
 	comm_title = "CO"
```
Comment (Rohesie) [inline code/game/jobs/job/shipside.dm]:
```suggestion
	comm_title = "CPT"
```

File: code/modules/clothing/under/marine_uniform.dm [hunk 6]
```diff
@@ -132,6 +132,15 @@
...
 	..()
 
+/obj/item/clothing/under/marine/officer/field
+	name = "executive officer uniform"
+	desc = "A uniform typically worn by TGMC Field Officers. It has shards of light Kevlar to help protect against stabbing weapons, bullets, and shrapnel from explosions, a small EMF distributor to help null energy-based weapons, and a hazmat chemical filter weave to ward off biological and radiation hazards."
+	icon_state = "BO_jumpsuit" //TBD: Unique Field Officer Jumpsuit
+
+/obj/item/clothing/under/marine/officer/exec/New()
+	select_gamemode_skin(/obj/item/clothing/under/marine/officer/field)
+	..()
```
Comment (Rohesie) [inline code/modules/clothing/under/marine_uniform.dm]:
```suggestion
	return ..()
```

File: maps/Z.03.TGS_Theseus.dmm [hunk 7]
```diff
@@ -2047,7 +2047,7 @@
 "aNs" = (/obj/structure/table/almayer,/obj/item/device/camera,/turf/open/floor/wood,/area/almayer/living/bridgebunks)
 "aNt" = (/obj/structure/table/almayer,/obj/item/paper_bin,/obj/item/tool/pen,/turf/open/floor/wood,/area/almayer/living/bridgebunks)
 "aNu" = (/obj/structure/table/almayer,/obj/item/device/flashlight/lamp,/obj/item/tool/crowbar,/turf/open/floor/wood,/area/almayer/living/bridgebunks)
-"aNv" = (/obj/structure/bed,/obj/effect/landmark/start{name = "Staff Officer"},/obj/item/bedsheet/brown,/turf/open/floor/wood,/area/almayer/living/bridgebunks)
```
Comment (Rohesie) [inline maps/Z.03.TGS_Theseus.dmm]: Do staff officers get landmarks now?

File: code/game/jobs/job/marines.dm [hunk 8]
```diff
@@ -139,6 +139,25 @@ You can serve a variety of roles, so choose carefully."}
 	H.equip_to_slot_or_del(new /obj/item/clothing/head/helmet/specrag(H), WEAR_HEAD)
 
 
+/datum/job/marine/smartgunner
```
Comment (Rohesie) [inline code/game/jobs/job/marines.dm]: Why is there a smartgunner datum being added? Wasn't there one already?

**Comment by spookydonut** [comment]
on the fly rank changes can cause confusion for players.

**Comment by Rohesie** [comment]
Let the wiki people wiki, and let's not demand the coders do that.

**Comment by DominikPanic** [comment]
Eh, it doesn't need to be complicated, could be just plaintext. Right now we are using CM's old system which everyone knows. If you plan to change that, you should explain the changes so people know how to RP and who orders who.

**Comment by Rohesie** [comment]
There's the changelog and the PR body to explain every change.

**Comment by DominikPanic** [comment]
It doesn't tell me anything about the structure.

**Comment by DominikPanic** [comment]
There's a new PR up now doing the same thing basically, feel free to re-open this if you start work on it again. ^^ #539

**Pull request closed without merging**